### ISSUE TITLE
Delay in processing orders

### ISSUE THREAD

**Issue by Pritesh-Mhatre** [Pritesh-Mhatre/autotrader#107]
Reporters: pandianr2005@gmail.com, nagaraja.hr@gmail.com

Sometimes AutoTrader takes time to process the commands read from orders.csv file. It is difficult to reproduce as it does not happen every time.

**Progress so far:**

- Added additional logs to pin point the code which is causing delay
- Found the delay section of the code, but there wasn't anything wrong in the code as such
- Tried a different mechanism of loading file eagerly, but even that did not help
- From one user's logs it seemed that "Platform Sync" might be interfering but that wasn't the case in another user's logs
- There is nothing obvious which catches the eye in the code, so we are not trying to reproduce it in our test machine in order better understand and investigate further

**Comment by Pritesh-Mhatre** [comment]
The problem is not reading the file. The issue is once reading is complete, execution halts for some time before processing lines from the file.

**Comment by Pritesh-Mhatre** [comment]
We did a lot of testing new way of processing lines, but not even once we saw delay. Now the only way to make any progress; on this is to ask the users to run different version of processing code.
We will send new code to users and wait for their findings.

**Comment by Pritesh-Mhatre** [comment]
We have checked everything, there is only 1 likely issue.
Getting database connection might take time sometimes.

Now the fix for above issue is not an easy one as it would require modifying critical components of AutoTrader. We cannot take this risk at the moment.

The new proposed **AutoTraderWeb** software which we are working on will completely change the way orders are saved. So this issue may not happen in newer version and even if it does happen; we will be able to investigate it because of enhanced and simpler architecture of the the new software.

We have closed the issue for now, we can reopen it when **AutoTraderWeb** goes live in 6 months time.
